path.go

```diff
@@ -30,6 +30,24 @@ func FromCid(c cid.Cid) Path {
 	return Path("/ipfs/" + c.String())
 }
 
+// ToCid attempts to convert the Path to a CID. This should always work if
+// the path was created from ParsePath.
+func (p Path) ToCid() (cid.Cid, error) {
+	parts := strings.Split(p.String(), "/")
+	if len(parts) < 3 || parts[2] == "" {
+		return cid.Undef, &pathError{fmt.Errorf("not enough path components"), p.String()}
+	}
+
+	switch parts[1] {
+	// TODO: Silently allowing IPNS though ParsePath doesn't validate the CID
+	//  in that case.
```

Comment:
Note this. I'm not sure what's the expected IPNS format; we might need to validate its CID either way.